code/ATMOSPHERICS/datum_pipe_network.dm

```diff
@@ -11,91 +9,83 @@ datum/pipe_network
 	var/update = 1
 	//var/datum/gas_mixture/air_transient = null
 
-	New()
-		//air_transient = new()
-
-		..()
+/*
+/datum/pipe_network/New()
+	//air_transient = new()
 
-	Destroy()
-		STOP_PROCESSING(SSmachines, src)
+	..()*/
 
-		for(var/obj/machinery/atmospherics/normal_member in normal_members)
-			normal_member.reassign_network(src,null)
+/datum/pipe_network/Process()
+	//Equalize gases amongst pipe if called for
+	if(update)
+		update = 0
+		reconcile_air() //equalize_gases(gases)
 
-		for(var/datum/pipeline/line_member in line_members)
-			line_member.network = null
+	//Give pipelines their process call for pressure checking and what not. Have to remove pressure checks for the time being as pipes dont radiate heat - Mport
+	//for(var/datum/pipeline/line_member in line_members)
+	//	line_member.process()
 
-		for(var/datum/gas_mixture/gas in gases)
-			gas.networks -= src
+/datum/pipe_network/proc/build_network(obj/machinery/atmospherics/start_normal, obj/machinery/atmospherics/reference)
+	//Purpose: Generate membership roster
+	//Notes: Assuming that members will add themselves to appropriate roster in network_expand()
 
-		gases.Cut()
-		normal_members.Cut()
-		line_members.Cut()
+	if(!start_normal)
+		qdel(src)
 
-		. = ..()
+	start_normal.network_expand(src, reference)
 
-	Process()
-		//Equalize gases amongst pipe if called for
-		if(update)
-			update = 0
-			reconcile_air() //equalize_gases(gases)
+	update_network_gases()
 
-		//Give pipelines their process call for pressure checking and what not. Have to remove pressure checks for the time being as pipes dont radiate heat - Mport
-		//for(var/datum/pipeline/line_member in line_members)
-		//	line_member.Process()
+	if((normal_members.len>0)||(line_members.len>0))
+		START_PROCESSING(SSmachines, src)
```

Comment:
usage of ssmachines for processing pipenets gives me anxiety, but i believe it should function, given how pipenets functioned before